pkg/protoassert/assert.go

```diff
@@ -40,6 +40,16 @@ func SlicesEqual[T proto.Message](t testing.TB, expected, actual []T, msgAndArgs
 	return areEqual
 }
 
+func SliceContains[T proto.Message](t testing.TB, slice []T, element T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	for _, e := range slice {
+		if proto.Equal(e, element) {
+			return true
+		}
+	}
+	return assert.Failf(t, "Slice does not contain element: %s %v", element.String(), msgAndArgs)
```

Comment:
The signature of `Failf` is:
`func Failf(t TestingT, failureMessage string, msg string, args ...interface{}) bool`

I think you need a generic message as first argument, and not a format string.

Tested locally, the following code:
```
package test

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFailf(t *testing.T) {
	assert.Failf(t, "message template: %s", "hello")
}
```
gives this result:
```
=== RUN   TestFailf
    tst_test.go:10: 
        	Error Trace:	/path/to/tst_test.go:10
        	Error:      	message template: %s
        	Test:       	TestFailf
        	Messages:   	hello
--- FAIL: TestFailf (0.00s)
```

Reply:
You're right, fixed

```go
package protocompat

import (
	"testing"

	"github.com/stackrox/rox/pkg/protoassert"
)

func TestEmpty(t *testing.T) {
	refEmpty := &Empty{}
	protoassert.SliceContains(t, []*Empty{}, refEmpty, "some message", "and", "args")
}

```

```
        	Error:      	Slice does not contain element
        	Test:       	TestEmpty
                Messages:   	"" [some message and args]
```